operator-common/src/main/java/io/strimzi/operator/common/Annotations.java

```diff
@@ -55,6 +55,12 @@ public static int intAnnotation(PodTemplateSpec podSpec, String annotation, int
         return str != null ? parseInt(str) : defaultValue;
     }
 
+    public static boolean hasAnnotation(HasMetadata resource, String annotation) {
+        ObjectMeta metadata = resource.getMetadata();
+        String str = annotation(annotation, null, metadata, null);
```

Comment:
Won't this NPE if the annotation is missing because you're passing a null argument for `deprecatedAnnotations`? Likely we need a null check on line 91, but perhaps switching from `String[] deprecatedAnnotations` to `String...` for the parameter declaration would work out nicer.